src/commands/register.ts

```diff
@@ -278,6 +278,7 @@ export default class ComponentRegister extends BaseCommand {
         continue;
       }
       delete service.debug; // we don't need to compare the debug block for remotely-deployed components
+      delete service.build; // build block data isn't relevant to remotely-deployed components
```

Comment:
I want to keep this so we can understand how someone built their component for debugging purposes. There are times when someone asks for help I go and check their spec.

Reply:
Allowing this to be included would be registering an invalid spec because we insert an `image` property to a service spec when/if an image is built. When this gets merged, using the combination of `build` and `image` should result in an error from the API. Do you have any thoughts on how you'd like to implement a workaround?

Reply:
That is a valid point. I think making a ticket for debug info that gets submitted alongside the spec may be useful, but don't want to block this.